### DISCUSSION TITLE
How to stop hands free.

### DISCUSSION THREAD

**Discussion by waymorphood** [speechly/speechly#220] [Q&A] [answered]
I see in the demo NodeJS app: "Set vad.enable to true for hands free use". What is the proper way to stop this?  I tried calling stop(), but it doesn't seem to work

**Comment by arzga** [comment, marked as answer]
Hi! After initializing BrowserClient with `vad` parameter present (this prepares the VAD engine – I think you're doing this already), you can toggle VAD on/off on-the-fly like so:

`client.adjustAudioProcessor( { vad: { enabled: true }})` and `client.adjustAudioProcessor( { vad: { enabled: false }})`

Setting `enabled: false` also ensures audio processing is stopped, which is probably what you need.

When VAD is enabled, it takes control of calling `BrowserClient.start()`and `stop()`. The effect of manually calling these with VAD enabled is currently unspecified.

**Reply by waymorphood** [reply]
Thanks for the answer! I am trying to shut down the recognition. So I first need to set the vad:{enabled:false} then call stop?

**Reply by arzga** [reply]
Calling `stop()` is not necessary. Setting `vad: {enabled: false}` will call it automatically if it's still listening.

**Reply by waymorphood** [reply]
cool. that works!
Needed to close the microphone for the browser to see it has been released. 
Thanks Ari!

Follow up question. I need to pause recognition when I am using TTS. Other than stopping/starting, is there a better way?

**Reply by aukkonen** [reply]
@waymorphood
Sorry, I only now saw your follow up question. Calling stop/start explicitly is probably your best option to avoid having TTS voices "leak" to our API.
